CMakeLists.txt

```diff
@@ -125,7 +125,8 @@ if(${ARBORX_ENABLE_TESTS} OR ${ARBORX_ENABLE_EXAMPLES})
   enable_testing()
 endif()
 
-if(${ARBORX_ENABLE_TESTS})
+cmake_dependent_option(ARBORX_ENABLE_HEADER_SELF_CONTAINMENT_TESTS "Enable header self-containment unit tests" ON "ARBORX_ENABLE_TESTS" OFF)
+if(ARBORX_ENABLE_HEADER_SELF_CONTAINMENT_TESTS)
```

Comment:
You could move this `if()` inside the `if(ARBORX_ENABLE_TESTS)` for clarity, but it's up to you.

Reply:
Decided not to but not opposed to change later.